internal/kuberesource/sets.go

```diff
@@ -627,3 +627,159 @@ sleep inf
+	cryptsetup luksFormat --pbkdf-memory=10240 $device "${tmp_key_path}" </dev/null
+
+	# Now we can get the LUKS UUID and deterministically derive the disk encryption key from the workload secret.
+	uuid=$(blkid "${device}" -s UUID -o value)
+	openssl kdf -keylen 32 -kdfopt digest:SHA2-256 -kdfopt key:$(cat "${workload_secret_path}") \
+		-kdfopt info:${uuid} -binary HKDF | base64 | tr -d '\n' > "${disk_encryption_key_path}"
+	cryptsetup luksChangeKey --pbkdf-memory=10240 "${device}" --key-file "${tmp_key_path}" "${disk_encryption_key_path}"
+	cryptsetup open "${device}" state -d "${disk_encryption_key_path}"
+	mkfs.ext4 /dev/mapper/state
+	cryptsetup close state
+fi
+
+# No matter if this is the first initialization derive the key (again) and open the device.
+uuid=$(blkid "${device}" -s UUID -o value)
+openssl kdf -keylen 32 -kdfopt digest:SHA2-256 -kdfopt key:$(cat "${workload_secret_path}") \
+	-kdfopt info:${uuid} -binary HKDF | base64 | tr -d '\n' > "${disk_encryption_key_path}"
+cryptsetup luksUUID "${device}"
+cryptsetup open "${device}" state -d "${disk_encryption_key_path}"
+mkdir -p /data
+mount /dev/mapper/state /srv/state
+touch /done
+sleep inf
+`
+
+	backend := StatefulSet("mysql", "").
+		WithSpec(StatefulSetSpec().
+			WithPersistentVolumeClaimRetentionPolicy(applyappsv1.StatefulSetPersistentVolumeClaimRetentionPolicy().
+				WithWhenDeleted(appsv1.DeletePersistentVolumeClaimRetentionPolicyType).
+				WithWhenScaled(appsv1.DeletePersistentVolumeClaimRetentionPolicyType)).
+			WithReplicas(1).
+			WithSelector(LabelSelector().
+				WithMatchLabels(map[string]string{"app.kubernetes.io/name": "mysql"}),
+			).
+			WithServiceName("mysql-svc").
+			WithTemplate(PodTemplateSpec().
+				WithLabels(map[string]string{"app.kubernetes.io/name": "mysql"}).
+				WithSpec(
+					PodSpec().
+						WithInitContainers(
+							Container().
+								WithName("luks-setup").
+								WithImage("ghcr.io/edgelesssys/contrast/cryptsetup:latest").
+								WithCommand("/bin/sh", "-c", initCommand).
+								WithVolumeDevices(
+									applycorev1.VolumeDevice().
+										WithName("state").
+										WithDevicePath("/dev/csi0"),
+								).
+								WithVolumeMounts(
+									VolumeMount().
+										WithName("share").
+										WithMountPath("/srv/state").
+										WithMountPropagation(corev1.MountPropagationBidirectional),
+									VolumeMount().
+										WithName("contrast-secrets").
+										WithMountPath("/contrast"),
+								).
+								WithSecurityContext(
+									applycorev1.SecurityContext().
+										WithPrivileged(true),
+								).
+								WithResources(ResourceRequirements().
+									WithMemoryLimitAndRequest(100),
+								).
+								WithStartupProbe(
+									Probe().
+										WithFailureThreshold(20).
+										WithPeriodSeconds(5).
+										WithExec(applycorev1.ExecAction().
+											WithCommand("/bin/sh", "-c", "cat /done"),
```

Comment:
```suggestion
											WithCommand("/bin/test", "-f", "/done"),
```
... might be clearer.